src/Courses/CoursePage.tsx

```diff
@@ -1,22 +1,51 @@
 import React, { useEffect, useState } from 'react';
 import CourseList from './CourseList';
+import AxiosRequest from '../Hooks/AxiosRequest';
+import { map } from 'lodash';
+import { CourseObject } from './CourseInterfaces';
+import { Container, Row, Button, Col } from 'react-bootstrap';
+import { BsPlusSquare } from 'react-icons/bs';
+import Cookies from 'js-cookie';
 
 interface CoursePageProps {
 
 }
 
 export const CoursePage: React.FC<CoursePageProps> = () => {
-    const [courses, setCourses] = useState<Array<String>>([]);
+    const [courses, setCourses] = useState<Array<CourseObject>>([]);
+    const userType = Cookies.get('userType');
 
     // Get the list of courses to render.
     useEffect(() => {
-        setCourses(['Course 1', 'Course 2', 'Course 3']);
+        (async () => {
+            try {
+                let res = await AxiosRequest.get('/courses');
+                console.log(res.data.data);
+                const courses: Array<CourseObject> = map(res.data?.data, obj => new CourseObject(obj));
+
+                setCourses(courses);
+            } catch (e) {
+                console.log(e.response);
+                setCourses([]);
+            }
+        })();
     }, []);
 
     return (
         <div>
-            <h1>My Courses</h1>
-            <CourseList courses={courses}/>
+            <Container>
+                <Row>
+                    <Col md={10}>
+                        <h1>My Courses</h1>
+                    </Col>
+                    <Col md={2}>
+                        {userType === 'Professor' && (
```

Comment:
We should have user type in an enum and compare that

Reply:
I'll handle this in a new PR- I have to actually hook in the role, and right now I'm just hard-coding it.